### PR TITLE
Spotify api working htmx

### PR DESCRIPTION
**Pull request by abaddonpuff**
HTMX is now working and shows the dropbox
image

### CODE REVIEW

**Comment by bbelderbos** [comment]
WOW cool, amazing!

File: webCommander/centralHub/xTwitter/twitterAPI.py [hunk 1]
```diff
@@ -44,6 +45,8 @@ def x_authenticate():
 def request_to_x_api(endpoint, headers={}, params={}):
     x_session = x_authenticate()
     response = x_session.get(endpoint, headers=headers, params=params)
+
+    print(f'Request to API {response.status_code}')
     while response.status_code == TOO_MANY_REQUESTS:
         print('Too many requests, trying in 15 mins...')
         sleep(900)
```
Comment (bbelderbos) [inline webCommander/centralHub/xTwitter/twitterAPI.py:52]: this is a pretty big timeout, I think the web server will time out before then:

image

File: webCommander/centralHub/views.py [hunk 2]
```diff
@@ -36,3 +37,17 @@ def handle_summary(request, twitter_handle):
...
+    allartists = SpotifyArtistInfo.objects.all()
+
+    return render(request, 'spotifyFramework/spotify_summary.html',{'allartists':allartists})
+
+def get_artists(request):
+    name = request.GET.get("artist", "")
+    response = search_spotify_artist(name)
+
+    options = ''.join(
+        "<option value='{val}'>{val}, Popularity: {pop}</option>".format(val=row[0], pop=row[2]) for row in response
```
Comment (bbelderbos) [inline webCommander/centralHub/views.py:51]: not sure why I suggested format here, nothing wrong with it, but you can also use an f-string :)

File: webCommander/centralHub/views.py [hunk 3]
```diff
@@ -36,3 +37,17 @@ def handle_summary(request, twitter_handle):
 
 def home_page(request):
     return render(request, 'home.html')
+
+def list_artist(request):
+    allartists = SpotifyArtistInfo.objects.all()
+
+    return render(request, 'spotifyFramework/spotify_summary.html',{'allartists':allartists})
+
+def get_artists(request):
```
Comment (bbelderbos) [inline webCommander/centralHub/views.py:46]: do we want to add @login_required to make it only for logged in users? so nobody goes DOS attack by making many requests to the Spotify API exhausting the API rate limits?

File: webCommander/centralHub/templates/spotifyFramework/spotify_summary.html [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+{% load bootstrap5 %}
+{% block content %}
+    <h1>Available Artists</h1>
+
+    <input id="artist" name="artist" type="text" placeholder="Search for an artist" hx-get="/artists" hx-trigger="submit_artist" hx-target="#artistChoices"  hx-indicator=".htmx-indicator">
+    <button type="button" onclick="document.querySelector('#artist').dispatchEvent(new Event('submit_artist'))">Submit</button>
+    <br>
+
+    <select id="artistChoices" name="artistChoices"
+      hx-post="/submit-artist">
```
Comment (bbelderbos) [inline webCommander/centralHub/templates/spotifyFramework/spotify_summary.html:11]: This is yet to be implemented and the next step right?

File: webCommander/centralHub/templates/base.html [hunk 5]
```diff
@@ -5,11 +5,12 @@
     <title>{% block title %}{% endblock %}</title>
     {% bootstrap_css %}
     {% bootstrap_javascript %}
+    <script src="https://unpkg.com/htmx.org@1.9.12"></script>
 </head>
 <body class="d-flex flex-column min-vh-100">
   <nav class="navbar navbar-expand-lg navbar-light bg-light">
     <div class="container-fluid">
-      <a class="navbar-brand" href="#">CommandCenter</a>
+      <a class="navbar-brand" href="/">CommandCenter</a>
```
Comment (bbelderbos) [inline webCommander/centralHub/templates/base.html:13]: I try to use {% url ... %} for each href in templates

File: webCommander/centralHub/spotify/spotifyAPI.py [hunk 6]
```diff
@@ -48,35 +52,59 @@ def get_spotify_artist(artist_name):
...
+
+    artist_id = search_spotify_artist(artist_name)
+
+    url = SPOTIFY_BASE_URL + SPOTIFY_ARTISTS + artist_id + '/albums'
+
+    response = requests.get(url, headers=spotify_authenticate())
+    albums_results = json.loads(response.text)
+
+    #TODO
+    #not sure if this is the best data type for this purpose
```
Comment (bbelderbos) [inline webCommander/centralHub/spotify/spotifyAPI.py:98]: is this ok now or do we need a refactoring here? let me know if you have a question/doubt here ...

**Review by bbelderbos — APPROVED** [review]
Awesome! Need to submit yet right? 
What are you working on today? 
Questions/ doubts?

File: webCommander/centralHub/xTwitter/twitterAPI.py (hunk 1 above)
Comment (abaddonpuff) [reply webCommander/centralHub/xTwitter/twitterAPI.py:52]: Uhh how would you think we should handle, maybe telling the user to try again in 15 minutes and start a counter to tell when should he try again? it takes 15 minutes for the API if there are too many requests to being able to query again.

**Comment by abaddonpuff** [comment]
> Awesome! Need to submit yet right? What are you working on today? Questions/ doubts?

Today I'm working on inserting the results on the DB. And will begin working on the league of legends one. Will make up for the lost time this week on the weekend.

File: webCommander/centralHub/templates/spotifyFramework/spotify_summary.html (hunk 4 above)
Comment (abaddonpuff) [reply webCommander/centralHub/templates/spotifyFramework/spotify_summary.html:11]: Correct, This is the next step

File: webCommander/centralHub/spotify/spotifyAPI.py (hunk 6 above)
Comment (abaddonpuff) [reply webCommander/centralHub/spotify/spotifyAPI.py:98]: This is OK. Will remove the tag on the next PR. I think its working very well.

File: webCommander/centralHub/xTwitter/twitterAPI.py (hunk 1 above)
Comment (bbelderbos) [reply webCommander/centralHub/xTwitter/twitterAPI.py:52]: Yes I think that is inevitable, or implement 'polling' with say htmx ...

File: webCommander/centralHub/templates/spotifyFramework/spotify_summary.html (hunk 4 above)
Comment (bbelderbos) [reply webCommander/centralHub/templates/spotifyFramework/spotify_summary.html:11]: ok I am seeing a PR, will review ...

**Pull request merged**